dolib/managers/registry.py

```diff
@@ -71,28 +61,92 @@ def repository_tags(
 
         return [models.Registry.Repository.Tag(**tag) for tag in res]
 
-    def delete_tag(
-        self, name: str = None, repository_name: str = None, tag: str = None
-    ) -> None:
-        assert name is not None, "name must be set"
-        assert repository_name is not None, "repository_name must be set"
-        assert tag is not None, "tag must be set"
+    def delete_tag(self, name: str, repository_name: str, tag: str) -> None:
+        self._client.request(
+            endpoint="registry/{name}/repositories/{repository_name}/tags/{tag}".format(
+                name=name, repository_name=repository_name, tag=tag
+            ),
+            method="delete",
+        )
 
+    def delete_manifest(self, name: str, repository_name: str, manifest: str) -> None:
         self._client.request(
+            endpoint="registry/{name}/repositories/{r_name}/digests/{manifest}".format(
+                name=name, r_name=repository_name, manifest=manifest
+            ),
+            method="delete",
+        )
+
+
+class AsyncRegistryManager(AsyncBaseManager):
+    endpoint: str = "registry"
+    name: str = "registry"
+
+    async def get(self) -> models.Registry:
+        res = await self._client.request(endpoint="registry", method="get",)
+        return models.Registry(**res["registry"])
+
+    async def create(self, registry: models.Registry) -> models.Registry:
+        res = await self._client.request(
+            endpoint="registry", method="post", data=registry.json(include={"name"}),
+        )
+        return models.Registry(**res["registry"])
+
+    async def delete(self, registry: models.Registry) -> None:
+        # may be in future account will be have multiple repository
+        await self._client.request(
+            endpoint="registry", method="delete",
+        )
+
+    async def docker_credentials(
+        self, read_write: bool = None, expiry_seconds: int = None
```

Comment:
```suggestion
        self, read_write: Optional[bool] = None, expiry_seconds: Optional[int] = None
```

I wonder why didn't `mypy` capture this?

Reply:
Oh, I see. That's because: https://github.com/geraxe/dolib/blob/ab8635de53b93f9fd94b06e1d3a64936f8ae4fd1/setup.cfg#L58

Reply:
Personally I'd prefer explicit optional, but it's out of the scope.